OpenSim/Common/Reporter.h

```diff
@@ -177,7 +192,17 @@ OpenSim_DECLARE_CONCRETE_OBJECT_T(TableReporter_, InputT, Reporter<InputT>);
               const auto& value = chan.getValue(state);
               result[idx] = value;
         }
-        const_cast<Self*>(this)->_outputTable.appendRow(state.getTime(), result);
+        try {
+            const_cast<Self*>(this)->_outputTable.appendRow(state.getTime(),
+                                                            result);
+        } catch(const InvalidTimestamp& exception) {
+            OPENSIM_THROW(Exception,
+                          std::string{exception.what()} +
```

Comment:
Minor comment, but it may be easier for the user to read/react to the main message first and append the actual lower-level exception as "details".